File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/person/MockablePersonRepository.java
package io.quarkus.it.mongodb.panache.person;

import java.util.List;

import jakarta.enterprise.context.ApplicationScoped;

import io.quarkus.mongodb.panache.PanacheMongoRepositoryBase;
import io.quarkus.panache.common.Sort;

@ApplicationScoped
public class MockablePersonRepository implements PanacheMongoRepositoryBase<PersonEntity, Long> {
    public List<PersonEntity> findOrdered() {
        return findAll(Sort.by("lastname", "firstname")).list();
    }
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/person/MockablePersonRepository.kt
package io.quarkus.it.mongodb.panache.person

import io.quarkus.mongodb.panache.kotlin.PanacheMongoRepositoryBase
import io.quarkus.panache.common.Sort
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
open class MockablePersonRepository : PanacheMongoRepositoryBase<PersonEntity, Long> {
    open fun findOrdered(): List<PersonEntity>? {
        val sort = Sort.by("lastname", "firstname")
        val found = findAll(sort)
        return found.list()
    }
}
